### PR TITLE
login: cleanups for enum InhibitWhat

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/login/logind-dbus.c [hunk 1]
```diff
@@ -1507,7 +1507,7 @@ static int execute_shutdown_or_sleep(
         int r;
 
         assert(m);
-        assert(w > 0);
+        assert(w >= 0);
```
Comment (keszybz) [inline src/login/logind-dbus.c]: I think `> 0` is valid here. This function should only be called if there's actually something to do.

Comment (yuwata) [reply src/login/logind-dbus.c]: Yeah, maybe so. But I have not followed all the possible path... I'd like to leave that for later cleanups.

Comment (keszybz) [reply src/login/logind-dbus.c]: It's only called from two places, and both exit early if InhibitWhat is 0.

Comment (yuwata) [reply src/login/logind-dbus.c]: OK, I will take a look later.

Comment (yuwata) [reply src/login/logind-dbus.c]: I confirmed that and dropped this part.

**Comment by yuwata** [comment]
I've dropped `assertion(w >= 0)` part as @keszybz suggested. I will change the tag to 'good-to-merge/waiting-for-ci'.

**Pull request merged**